src/components/MainContent/MainContent.tsx

```diff
@@ -0,0 +1,266 @@
+  };
+
+  const changeTodoOnServer = async (
+    id: number,
+    todo: Partial<Todo>,
+  ) => {
+    try {
+      addOnLoadingTodo(id);
+      await editTodo(id, todo);
+      await loadTodosFromServer();
+      deleteFromLoadingTodo(id);
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to update a todo');
+    }
+  };
+
+  const changeCompletedForGroup = async (group: Todo[], value: boolean) => {
+    group.map(todo => addOnLoadingTodo(todo.id));
+    await Promise.all(
+      todos.map(todo => editTodo(todo.id, { completed: value })),
+    );
+    await loadTodosFromServer();
+    todos.map(todo => deleteFromLoadingTodo(todo.id));
+  };
+
+  const setAllTodosCompletedOnServer = async () => {
+    try {
+      if (isAllTodosCompleted) {
+        await changeCompletedForGroup(completedTodos, false);
+      } else {
+        await changeCompletedForGroup(activeTodos, true);
+      }
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Can\'t change todos');
+    }
+  };
+
+  const deleteAllCompletedTodos = async () => {
+    try {
+      await Promise.all(
+        completedTodos.map(todo => deleteTodoOnServer(todo.id)),
+      );
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Can\'t delete todos');
+    }
+  };
+
+  // Effects
+  useEffect(() => {
+    loadTodosFromServer();
+  }, []);
+
+  useEffect(() => {
+    if (tempTodo) {
+      addNewTodoOnServer(tempTodo);
+      setIsTodoAdding(false);
+    }
+  }, [tempTodo]);
+
+  return (
+    <div className="todoapp__content">
+      <header className="todoapp__header">
+        <ToggleTodosCompleted
+          setTodosCompleted={setAllTodosCompletedOnServer}
+          isToggleVisible={isToggleVisible}
+        />
+
+        <CreateTodoForm
+          addNewTodo={addTempTodo}
+          isTodoAdding={isTodoAdding}
+        />
+      </header>
+
+      <section className="todoapp__main" data-cy="TodoItem">
+        {filteredTodos().map(todo => {
+          return (
+            <TodoItem
+              key={todo.id}
+              todo={todo}
+              deleteTodo={deleteTodoOnServer}
+              isLoadingTodo={loadingTodos.includes(todo.id)}
+              changeTodoOnServer={changeTodoOnServer}
+            />
+          );
+        })}
+      </section>
+
+      {todos.length > 0 && (
+        <footer className="todoapp__footer" data-cy="Footer">
+          <span className="todo-count" data-cy="todosCounter">
+            {`${activeTodos.length} items left`}
+          </span>
+
+          <nav className="filter" data-cy="Filter">
+            {filterList.map((filter, index) => {
+              const newKey = filter + index;
```

Comment:
you shouldn't use index in process of generating key for the elements
in this way the key will be different if one of elements will be deleted, and it won't help React to optimize rendering for such elements